collectors/python.d.plugin/am2320/am320.chart.py

```diff
@@ -0,0 +1,60 @@
+try:
+    import board
+    import busio
+    import adafruit_am2320
+    HAS_STUFF = True
+except ImportError:
+    HAS_STUFF = False
+
+
+#from bases.FrameworkServices.SimpleService import SimpleService
+from base import SimpleService
+
+
+ORDER = [
+    'temperature',
+    'humidity',
+]
+
+CHARTS = {
+    'temperature': {
+        'options': [None, 'Temperature', 'celsius', 'temperature', 'am2320.temperature', 'line'],
+        'lines': [
+            ['temperature']
+        ]
+    },
+    'humidity': {
+        'options': [None, 'Relative Humidity', 'percentage', 'humidity', 'am2320.humidity', 'line'],
+        'lines': [
+            ['humidity']
+        ]
+    }
+}
+
+
+class Service(SimpleService):
+    def __init__(self, configuration=None, name=None):
+        SimpleService.__init__(self, configuration=configuration, name=name)
+        self.order = ORDER
+        self.definitions = CHARTS
+        self.am = None
+
+    def check(self):
+        if not HAS_STUFF:
+            self.error("following libraries are needed to get module working: board, busio, adafruit_am2320")
+            return False
+
+        try:
+            i2c = busio.I2C(board.SCL, board.SDA)
+            self.am = adafruit_am2320.AM2320(i2c)
+        except Exception as error:
+            self.error("error on creating I2C shared bus : {0}".format(error))
+            return False
+
+        return True
+
+    def get_data(self):
+        return {
+            'temperature': self.am.temperature,
+            'humidity': self.am.relative_humidity,
```

Comment:
Is ti at all possible for access of either of these attributes to throw an exception?  If so, they need to be wrapped in a proper try/except clause, and if not it should probably be documented that they don't in a comment above the code in question.

Reply:
> Is it at all possible for access of either of these attributes to throw an exception?

 - `OSError`
 - `RuntimeError`

> https://github.com/adafruit/Adafruit_CircuitPython_AM2320/blob/26c73be5d524517e3f019716d952004a8f4c4dcb/adafruit_am2320.py#L98-L137

Reply:
@tommybuck

```python
try:
    return ...
except (OSError, RuntimeError) as error:
    self.error(error)
    return None
```

Reply:
@tommybuck Wrapping the function body itself.  So the whole function would be:

```python
def get_data(self):
    try:
        return {
            'temperature': self.am.temperature,
            'humidity': self.am.relative_humidity,
        }
    except (OSError, RuntimeError) as error:
        self.error(error)
        return None
```